File: easy/Raing-Char/Racing-Char.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        int posAnt = 0;
        boolean first = true;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            // Process line of input Here
            int pos;
            char ch;
            int gatePos = line.indexOf('_');
            int checPos = line.indexOf('C');
            if (checPos>=0) pos=checPos;
                else pos=gatePos;
            if (!first) {
                if (pos>posAnt) ch='\\';
                else if (pos<posAnt) ch='/';
                    else ch = '|';
            }else {
                ch='|';
                first = false;
            }
            posAnt = pos;
            System.out.println(line.substring(0,pos)+ch+line.substring(pos+1));
        }
    }
}

File: easy/Racing-Char/Racing-Char.cs
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        int posAnt=0;
        bool first=true;
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            line = line.Trim();
            int pos;
            char ch;
            int gatePos = line.IndexOf('_');
            int checPos = line.IndexOf('C');
            if (checPos>=0) pos=checPos;
                else pos=gatePos;
            if (!first) {
                if (pos>posAnt) ch='\\';
                else if (pos<posAnt) ch='/';
                    else ch = '|';
            }else {
                ch='|';
                first = false;
            }
            posAnt = pos;
            System.Console.WriteLine(line.Substring(0,pos)+ch+line.Substring(pos+1));
            
        }
    }
}
